src/Storage/Sha1FileHashingService.php

```diff
@@ -80,8 +84,31 @@ private function normaliseFilesystem($fs)
             );
         }
 
-        $filesystem = Injector::inst()->get(sprintf('%s.%s', Filesystem::class, $fs));
-        return $filesystem;
+        if (!isset($this->filesystems[$fs])) {
+            $this->filesystems[$fs] = Injector::inst()->get(sprintf('%s.%s', Filesystem::class, $fs));
+        }
+
+        return $this->filesystems[$fs];
+    }
+
+    /**
+     * Get the matching key for thep provided Filesystem
+     * @param string|Filesystem $keyOrFs
+     * @return string
+     */
+    private function getFilesystemKey($keyOrFs)
+    {
+        if (is_string($keyOrFs)) {
+            return $keyOrFs;
+        }
+
+        foreach ([AssetStore::VISIBILITY_PUBLIC, AssetStore::VISIBILITY_PROTECTED] as $visibility) {
```

Comment:
If you're going to change your adapter settings, I don't think it's an unreasonable expectation that you should flush your settings as well. In most cases you'll have to do that anyway to have your new settings take affect.

In any case, if your changes are somehow causing the hash of your files to change, you'll be completely screwed any way, because your database won't know anything about those new hashes.